src/libraries/System.Private.CoreLib/src/System/Guid.cs

```diff
@@ -423,29 +423,15 @@ public static bool TryParseExact(ReadOnlySpan<char> input, [StringSyntax(StringS
             input = input.Trim();
 
             var parseResult = new GuidResult(GuidParseThrowStyle.None);
-            bool success = false;
-            switch ((char)(format[0] | 0x20))
-            {
-                case 'd':
-                    success = TryParseExactD(input, ref parseResult);
-                    break;
-
-                case 'n':
-                    success = TryParseExactN(input, ref parseResult);
-                    break;
-
-                case 'b':
-                    success = TryParseExactB(input, ref parseResult);
-                    break;
-
-                case 'p':
-                    success = TryParseExactP(input, ref parseResult);
-                    break;
-
-                case 'x':
-                    success = TryParseExactX(input, ref parseResult);
-                    break;
-            }
+            bool success = (char)(format[0] | 0x20) switch
```

Comment:
Is the cast to `char` actually necessary here? It should work without it and that would make smaller IL and avoid an extra zero extension that the JIT has to otherwise remove here

Reply:
@tannergooding hm. I've removed the cast but the codegen hasn't changed. But I guess it is still makes sense to remove it as useless

Reply:
I'd guess the JIT is recognizing it as unnecessary and removing it already; just something small that caught my eye as likely unneeded